### DISCUSSION TITLE
Allow passing animated values to expo-gl as uniforms

### DISCUSSION THREAD

**Discussion by sergeymorkovkin** [software-mansion/react-native-reanimated#1739] [Ideas]
Of course, this is not something directly related to reanimated, but rather a feature request. This would require cooperation from both expo and reanimated teams. 

The thing is Reanimated 2 claims to support worklets which execute on a UI thread. Unfortunately, expo-gl is only available on the JS thread and therefore we can't move animation-related code to UI. To make a long story short, there are only a couple of things which are missing from using reanimated with GL context: requestAnimationFrame inside worklets and expo-gl availability on the UI Thread.

What do you think?

**Comment by Szymon20000** [comment]
Unfortunately, I'm not convinced it would be impactful for people. In our opinion, Lottie covers most of the cases that (GL + animations) would cover. However, I would like to hear your arguments why It's not true.

**Reply by sergeymorkovkin** [reply]
We are trying to build a simple mobile poker. Even though animated API works, we are already pushing it's limits. Not mentioning resource consumption overhead from 1500+ animated nodes, bridge communication and element tree memory consumption. On modern devices it's okay, but lower-end Androids need more optimizations where expo-gl could be very helpful.

**Comment by sergeymorkovkin** [comment]
Also, what do you think about [this thread](https://github.com/expo/expo/issues/11267#issuecomment-782092028)?
